tests/test_mlflow_integration.py

```diff
@@ -0,0 +1,154 @@
+from sklearn import datasets
+from sklearn.ensemble import RandomForestClassifier
+from sklearn.model_selection import train_test_split
+
+import mlflow
+
+
+@pytest.fixture(scope="module")
+def setup_servers():
+    with subprocess.Popen([sys.executable, "-m", "mlflow", "server", "--port", "5000"]) as proc:
+        try:
+
+            mlflow_tracking_url = "http://127.0.0.1:5000"
+            uc_oss_url = "uc:http://127.0.0.1:8080"
+
+            mlflow.set_tracking_uri(mlflow_tracking_url)
+            mlflow.set_registry_uri(uc_oss_url)
+
+            yield mlflow_tracking_url
+        finally:
+            proc.terminate()
+
+
+def test_integration(setup_servers):
+    catalog = "unity"
+    schema = "default"
+    registered_model_name = "iris"
+    model_name = f"{catalog}.{schema}.{registered_model_name}"
+    mlflow.set_experiment("iris-uc-oss")
+    try:
+        mlflow.MlflowClient().get_registered_model(model_name)
+    except Exception as e:
+        e.args[0].startswith("NOT_FOUND")
+    else:
+        assert False, "Expected exception for missing model not raised"
+
+    X, y = datasets.load_iris(return_X_y=True, as_frame=True)
+    X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)
+
+    def build_model():
+        with mlflow.start_run():
+            # Train a sklearn model on the iris dataset
+            clf = RandomForestClassifier(max_depth=7)
+            clf.fit(X_train, y_train)
+            # Take the first row of the training dataset as the model input example.
+            input_example = X.iloc[[0]]
+            # Log the model and register it as a new version in UC.
+            mlflow.sklearn.log_model(
+                clf,
+                "model",
+                # The signature is automatically inferred from the input example and
+                # its predicted output.
+                input_example=input_example,
+                registered_model_name=model_name,
+            )
+
+    try:
+        build_model()
+    except Exception as e:
+        assert False, f"Unhandled exception while attempting to build a model: {e}"
+
+    model_version = 1
+    model_uri = f"models:/{model_name}/{model_version}"
+    rm_desc = "UC-OSS/MLflow Iris model"
+    mv_desc = "Version 1 of the UC-OSS/MLflow Iris model"
+
+    # Load the model and do some batch inference.
+    try:
+        # By specifying the UC OSS model uri, mlflow will make UC OSS
+        # REST API calls to retrieve the model
+        loaded_model = mlflow.pyfunc.load_model(model_uri)
+        predictions = loaded_model.predict(X_test)
+        iris_feature_names = datasets.load_iris().feature_names
+        result = pd.DataFrame(X_test, columns=iris_feature_names)
+        result["actual_class"] = y_test
+        result["predicted_class"] = predictions
+        assert result[:4] is not None
+    except Exception as e:
+        assert False, f"Unhandled exception while loading a model: {e}"
+
+    try:
```

Comment:
Similar as above - we could just let any issues bubble up naturally if the call to list artifacts fails with the server